### PR TITLE
Add geolocation scanner

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by wilsonpage** [comment]
@arcturus r?

**Comment by wilsonpage** [comment]
This patch adds a single 'scan' for virtual-beacons whenever `scanner.start()` is called. A followup patch will be to automatically perform scans when the users location changes by X meters.

File: src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package org.mozilla.magnet.net.scanner.geolocation;
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java]: nit:
Can we move this class outside the org.mozilla.magnet.net package?

Maybe org.mozilla.magnet.gps? Does that work for you?

File: src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import org.json.JSONObject;
+import org.mozilla.magnet.net.scanner.BaseScanner;
+import org.mozilla.magnet.net.scanner.MagnetScannerItem;
+
+public class ScannerGeolocation extends BaseScanner implements ConnectionCallbacks, OnConnectionFailedListener {
+    private final static String TAG = "ScannerGeolocation";
+    private final static String SCANNER_TYPE = "geolocation";
+    private final static String SLUG_BASE_URL = "https://tengam.org/";
+    private final static String API_SEARCH_URL = "https://tengam.org/content/v1/search/beacons/";
+    private final static int SCAN_RADIUS = 10;
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java]: m? km? tiny comment with the unity won't hurt anyone.

File: src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    public void stop() {
+        mGoogleApiClient.disconnect();
+    }
+
+    @Override
+    public void onConnected(Bundle connectionHint) {
+        Location location = LocationServices.FusedLocationApi.getLastLocation(mGoogleApiClient);
+        Log.d(TAG, "lat: " + location.getLatitude());
+        Log.d(TAG, "lon: " + location.getLongitude());
+        scan(location);
```
Comment (arcturus) [inline src/main/java/org/mozilla/magnet/net/scanner/geolocation/ScannerGeolocation.java]: Should we check the location accuracy before performing a scan?

location.getAccuracy()

https://developer.android.com/reference/android/location/Location.html#getAccuracy()

**Review by arcturus — COMMENTED** [review]
Looking pretty good, just worry about the accuracy.

**Review by arcturus — APPROVED** [review]
Granting review since the accuracy changes are going to come in a follow up.

**Pull request merged**